### ISSUE TITLE
Failure persistence file location for workspace member crates

### ISSUE THREAD

**Issue by haraldmaida** [AltSysrq/proptest#24]
I've just started using proptest. It looks great so far. The project I'm using it is organized as a Cargo workspace with several member crates. When using proptest in a member crate of a workspace the location of failure persistence file is not what I expected.

The project structure is like this:
```
my_workspace/
|-> Cargo.toml    #defines the workspace
|-> my_member_crate/
    |-> Cargo.toml
    |-> src/lib.rs
    |-> src/protocol.rs   #code with tests module
```
in this case the failure persistence file is located at:
`my_workspace/my_member_crate/my_member_crate/protocol.protest-regressions`

what I expected is:
`my_workspace/my_member_crate/protest-regressions/protocol.txt`

I have played around with the source code of test_runner. I found that the source_path is relative to the workspace root, e.g. my_member_crate/src/lib.rs. But the working directory when running the tests is the crate root, so my_member_crate in this case.

**Comment by AltSysrq** [comment]
> found that the source_path is relative to the workspace root, e.g. my_member_crate/src/lib.rs

Odd, what OS are you on? I thought `file!()` always gave an absolute path

**Comment by haraldmaida** [comment]
It all happens on Windows 10.
Tests are executed with Cargo test.
Rust itself is setup via rustup. Rust 1.22.1 stable

I thought I double check it and added a `println("{:?}", file!());` to one of my tests in a member crate and it prints out: "member_crate\\src\\my_mod.rs".

**Comment by AltSysrq** [comment]
Thanks for the info! It looks like this problem also affected non-workspace projects as well.

This has been fixed in version 0.3.4, manually verified on Windows 10 with both non-workspace and workspace configurations. Anyone affected by this bug will need to manually move any persistence files to the correct location, for example by running in cmd or PS:

```cmd
move my_member_crate\my_member_crate\somesource.proptest-regressions my_member_crate\proptest-regressions\somesource.txt
```

The exact location where the files were incorrectly placed was dependent on the working directory when the tests were run. This means that if you ran tests outside of cargo, it is theoretically possible to have multiple persistence files that belong in the same place. If this has happened, simply concatenating the respective files will merge them correctly.

**Comment by haraldmaida** [comment]
Unfortunately starting from proptest version 0.7 the exact same behavior as in the original issue is happening again. In version 0.5 and 0.6 everything works perfectly fine.

I'm on Windows 10. Rust 1.26.0 stable.

It makes no difference whether I call "cargo test" from the workspace directory or inside the directory of the member-crate.

If I can help somehow please let me know. Let me thank you for the great work on this crate. I really love to use proptest in my projects.

**Comment by AltSysrq** [comment]
Sorry about the regression, I'll take a look this weekend. I'll also see about adding some automated tests around it this time.

**Comment by AltSysrq** [comment]
This is fixed in 0.7.1, now including [full-stack tests for both platforms](https://github.com/AltSysrq/proptest/tree/master/test-persistence-location).
